### DISCUSSION TITLE
Uno wasm project needs to download xml files based on a PHP $_SESSION

### DISCUSSION THREAD

**Discussion by adampulley** [unoplatform/uno#8758] [General]
I've a wasm app that I've hit 2 issues with
1) It needs to download various xml files from the server and also AWS S3, but I'm hitting roadblocks on how to achieve this.  Mechanisms like HttpClient are not implemented in Uno and the only potential one that the docs mention seems to be RandomAccessStreamReference.CreateFromUri. However finding a working example of actually using this has so far illuded me. I wanted to ask if anyone knows if this is the best / only method I can use and if so if anyone knows of any resource on how to use it under uno?
2) My users running the wasm will be logged into our Wordpress based site before running the uno wasm app and I need some way to pass the app the URL's it should use for these XML files for the logged in user (different Wordpress users see different XML files). So I ideally need to use a PHP $_SESSION in some way for the app to connect to the website and request this information.  With HttpClient unsupported I'm wondering if I've hit a major roadblock on this - can anyone recommend an approach I could use?
Many thanks
Adam

**Comment by jeromelaban** [comment]
> 1. Mechanisms like HttpClient are not implemented in Uno

`HttpClient` is available and working properly (it's provided by the underlying .NET Framework), see [this article](https://github.com/unoplatform/uno/blob/master/doc/articles/howto-consume-webservices.md). Note that you're in a browser context, and CORS restrictions apply.

> 2. I need some way to pass the app the URL's it should use for these XML files for the logged in user

You can access the current browser url using [javascript interop](https://platform.uno/docs/articles/interop/wasm-javascript-1.html#invoke-javascript-code-from-c), or by using the `Application.OnLaunched` argument parameters which provide the original query string which launched the app.

**Reply by adampulley** [reply]
Thanks so much for the quick response Jerome and solutions, that’s great to hear!
